.tekton/tasks/component-app-name.yaml

```diff
@@ -0,0 +1,63 @@
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: component-app-name
+spec:
+  params:
+   - name: APP_FILE_NAME
+     description: This is the file which contains the argocd application names running as per the component.
+     default: "staging-app.yaml"
+  description: >-
+    This task is to find put which applicationset needs to be updated for the staging cluster argocd application
+  workspaces:
+   - name: infra-deployment-git
+   - name: shared-file-path
+  steps:
+    - name: get-component-app-name
+      image: image-registry.openshift-image-registry.svc:5000/openshift/python:3.8
+      workingDir: $(workspaces.infra-deployment-git.path)
+      env:
+        - name: APP_FILE_NAME
+          value: $(params.APP_FILE_NAME)
+      script: |
+        #!/usr/bin/env python3
+        import subprocess
+        import re
+        import os
+
+        cmd = "pip3 install pyyaml"
```

Comment:
If you move the script to a .py file, I would put that bit in the `script` section of the task.